### PR TITLE
Processing chain: read data from RESTful API

### LINKED ISSUE #637: In processing chain read data from api endpoint
introduce server/modules/processingChain/api/rApi

### PR DESCRIPTION
**Pull request by SteRiccio**
Resolves #637

### CODE REVIEW

File: server/modules/analysis/api/processingChainApi.js [hunk 1]
```diff
@@ -167,6 +167,24 @@ export const init = app => {
     },
   )
 
+  // ====== READ - Chain Data
+
```
Comment (minotogna) [inline server/modules/analysis/api/processingChainApi.js]: Create rChainApi and use prefix /survey/:surveyId/rChain/step

File: server/modules/analysis/manager/processingChainManager.js [hunk 2]
```diff
@@ -214,6 +214,14 @@ const _updateCalculationIndexes = async (user, surveyId, step, t) => {
 
 export { countChainsBySurveyId, fetchChainsBySurveyId, fetchChainByUuid } from '../repository/processingChainRepository'
```
Comment (minotogna) [inline server/modules/analysis/manager/processingChainManager.js]: create rChainManager

File: server/modules/analysis/repository/processingChainRepository.js [hunk 3]
```diff
@@ -60,6 +70,31 @@ export const fetchChainByUuid = async (surveyId, processingChainUuid, client = d
     dbTransformCallback,
   )
```
Comment (minotogna) [inline server/modules/analysis/repository/processingChainRepository.js]: create rChainRepository

File: server/modules/analysis/manager/processingChainManager.js [hunk 4]
```diff
@@ -214,6 +214,14 @@ const _updateCalculationIndexes = async (user, surveyId, step, t) => {
...
 
+export const fetchStepData = async (survey, cycle, stepUuid) => {
+  const surveyId = Survey.getId(survey)
+  let step = await ProcessingStepRepository.fetchStepSummaryByUuid(surveyId, stepUuid)
+  const calculations = await ProcessingStepCalculationRepository.fetchCalculationsByStepUuid(surveyId, stepUuid)
```
Comment (minotogna) [inline server/modules/analysis/manager/processingChainManager.js]: use ```ProcessingStepRepository.fetchStepsAndCalculationsByChainUuid```

File: server/modules/analysis/service/_rChain/rFunctions/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export { NA, dfVar, merge, setVar, sysTime } from './rFunctionsUtility'
+
+export { dbDisconnect, dbGetQuery, dbSendQuery, dbWriteTable, setConnection } from './rFunctionsDb'
+
+export { arenaStartTime, arenaEndTime, arenaInfo, arenaPersistCalculationScript } from './rFunctionsArena'
+
+export { arenaGet, arenaPost } from './rFunctionsArenaApi'
```
Comment (minotogna) [inline server/modules/analysis/service/_rChain/rFunctions/index.js]: omit the prefix rFunctions from the files

File: server/modules/analysis/service/processingChainService.js [hunk 6]
```diff
@@ -23,6 +26,11 @@ export {
   deleteCalculation,
 } from '../manager/processingChainManager'
```
Comment (minotogna) [inline server/modules/analysis/service/processingChainService.js]: add rChainService

File: server/modules/category/api/categoryApi.js [hunk 7]
```diff
@@ -132,6 +133,28 @@ export const init = app => {
     },
   )
```
Comment (minotogna) [inline server/modules/category/api/categoryApi.js]: if it's used in rChain, then add this to rChainApi and use /api/rChain/category

File: server/modules/analysis/api/processingChainApi.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/analysis/api/processingChainApi.js]: ok

File: server/modules/analysis/manager/processingChainManager.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/analysis/manager/processingChainManager.js]: ok

File: server/modules/analysis/repository/processingChainRepository.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingChainRepository.js]: ok

File: server/modules/analysis/service/_rChain/rFunctions/index.js (hunk 5 above)
Comment (SteRiccio) [reply server/modules/analysis/service/_rChain/rFunctions/index.js]: ok

File: server/modules/analysis/service/processingChainService.js (hunk 6 above)
Comment (SteRiccio) [reply server/modules/analysis/service/processingChainService.js]: ok

File: server/modules/category/api/categoryApi.js (hunk 7 above)
Comment (SteRiccio) [reply server/modules/category/api/categoryApi.js]: as you wish

File: server/modules/analysis/manager/processingChainManager.js (hunk 4 above)
Comment (SteRiccio) [reply server/modules/analysis/manager/processingChainManager.js]: but we need to fetch only one step, not all the steps...

File: server/modules/analysis/manager/rChainManager.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+import * as ProcessingStep from '@common/analysis/processingStep'
+
+import * as ProcessingStepRepository from '../repository/processingStepRepository'
+import * as ProcessingStepCalculationRepository from '../repository/processingStepCalculationRepository'
+import * as RChainRepository from '../repository/rChainRepository'
+
+export const fetchStepData = async (survey, cycle, stepUuid) => {
+  const surveyId = Survey.getId(survey)
+  let step = await ProcessingStepRepository.fetchStepSummaryByUuid(surveyId, stepUuid)
+  const calculations = await ProcessingStepCalculationRepository.fetchCalculationsByStepUuid(surveyId, stepUuid)
```
Comment (minotogna) [inline server/modules/analysis/manager/rChainManager.js]: use Promise.all

Comment (SteRiccio) [reply server/modules/analysis/manager/rChainManager.js]: ok

**Pull request merged**